extensions/amp-story-captions/0.1/amp-story-captions.js

```diff
@@ -17,6 +17,11 @@ import {createShadowRootWithStyle} from '../../amp-story/1.0/utils';
 const presetValues = ['default', 'appear'];
 
 export class AmpStoryCaptions extends AMP.BaseElement {
+  /** @override  */
+  static previewAllowed() {
```

Comment:
We should actually check whether the track is cached or not, otherwise we might be loading an uncached track and break privacy

Reply:
My understanding is that, in order to check the track's source, we have to look at the `video` element's `<track>` children and check their `src` attributes. However, we don't have access to this video element from within the static `previewAllowed()` of amp-story-captions. Alternatively, we could update `prerenderAllowed()` in amp-video and amp-audio to return false if the tracks are not cached. Is there an existing method we could use to confirm whether a track's `src` is a cached URL?